openedx/core/djangoapps/xblock/runtime/runtime.py

```diff
@@ -175,34 +182,26 @@ def __init__(
                 )
                 If user_id is ANONYMOUS_USER, the handler should execute without
                 any user-scoped fields.
-            authored_data_store: A FieldData instance used to retrieve/write
-                any fields with UserScope.NONE
-            student_data_store: A FieldData instance used to retrieve/write
-                any fields with UserScope.ONE or UserScope.ALL
+            student_data_mode: Specifies whether student data should be kept
+                in a temporary in-memory store (e.g. Studio) or persisted
+                forever in the database.
+            runtime_class: What runtime to use, e.g. BlockstoreXBlockRuntime
         """
         self.handler_url = handler_url
         self.id_reader = OpaqueKeyReader()
         self.id_generator = MemoryIdManager()  # We don't really use id_generator until we need to support asides
         self.runtime_class = runtime_class
-        self.authored_data_store = authored_data_store
-        self.field_data = SplitFieldData({
-            Scope.content: authored_data_store,
-            Scope.settings: authored_data_store,
-            Scope.parent: authored_data_store,
-            Scope.children: authored_data_store,
-            Scope.user_state_summary: student_data_store,
-            Scope.user_state: student_data_store,
-            Scope.user_info: student_data_store,
-            Scope.preferences: student_data_store,
-        })
-
+        self.authored_data_store = BlockstoreFieldData()
+        assert student_data_mode in (self.STUDENT_DATA_EPHEMERAL, self.STUDENT_DATA_PERSISTED)
+        self.student_data_mode = student_data_mode
+        self.student_data_loaded = WeakKeyDictionary()
         self._error_trackers = {}
 
     def get_runtime(self, user_id):
         # type: (int) -> XBlockRuntime
         return self.runtime_class(self, user_id)
 
-    def get_service(self, scope_ids, service_name):
+    def get_service(self, block, service_name):
```

Comment:
Isn't this a pretty major change in the contract?

Reply:
This is `XBlockRuntimeSystem`, not `XBlockRuntime`, so this particular method is not exposed outside of this module. It's a new API I added in the original blockstore runtime PR. Changing `XBlockRuntime.get_service()` would definitely be a no.